crates/bevy_reflect/src/reflect.rs

```diff
@@ -11,6 +11,22 @@ use std::{
 use crate::utility::NonGenericTypeInfoCell;
 pub use bevy_utils::AHasher as ReflectHasher;
 
+/// A simple enumeration of "kinds" of type, without any associated object.
+///
+/// A `ReflectKind` is obtained via [`Reflect::reflect_kind`].
+#[derive(Copy, Clone, Debug)]
+pub enum ReflectKind {
+    Struct,
+    TupleStruct,
+    Tuple,
+    List,
+    Array,
+    Map,
+    Enum,
+    Value,
+    Dynamic,
```

Comment:
I don't think `ReflectKind::Dynamic` is used anywhere, right? All the Dynamic types use their corresponding kind already from the looks of it. If so, could we remove this variant?